src/main/java/seedu/address/model/person/company/Company.java

```diff
@@ -54,6 +54,25 @@ public StudentId getStudentId() {
         return null;
     }
 
+    /**
+     * Returns true if both persons have the same company.
+     * This defines a weaker notion of equality between two companies.
+     */
+    @Override
+    public boolean isSamePerson(Person otherPerson) {
+        if (otherPerson == this) {
+            return true;
+        }
+
+        if (otherPerson instanceof Company) {
+            return otherPerson != null
+                    && otherPerson.getName().equals(getName())
+                    && otherPerson.getIndustry().equals(getIndustry());
+        }
+
+        return false;
+    }
+
     @Override
```

Comment:
Good update for refining the uniqueness in `isSamePerson()`. We might need to check the handling of companies with industry null values